### PR TITLE
Create pelikan rust bindings

### PR DESCRIPTION
**Pull request by swlynch99**
This is the second of 3 PRs for #260.

It has
- Complete (except for static inline functions) low-level bindings to all pelikan modules.
- Just enough high-level bindings to rewrite the pingserver in rust
- Formats cdb-rs using rustfmt

### CODE REVIEW

File: src/rust-util/pelikan/src/core/mod.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+use std::ffi::c_void;
+use std::mem::MaybeUninit;
+
+#[repr(C)]
+pub enum DataProcessorError {
+    Error = -1,
+}
+
+/// Methods for handling messages on the data port
+pub trait DataProcessor {
```
Comment (swlynch99) [inline src/rust-util/pelikan/src/core/mod.rs]: I'm not happy with how this turned out and I plan to rewrite this in a follow-on PR

File: src/storage/cdb/cdb_rs/src/cdb/ffi/mod.rs [hunk 2]
```diff
@@ -1,15 +1,14 @@
 use cc_binding as bind;
 use ccommon_rs::bstring::BStr;
-use cdb::{cdb_handle, Reader, Result};
 use cdb;
+use cdb::{cdb_handle, Reader, Result};
```
Comment (brayniac) [inline src/storage/cdb/cdb_rs/src/cdb/ffi/mod.rs]: assuming this is rustfmt change, but I'd probably do `cdb::{self, ....}`.

File: src/rust-util/pelikan/src/core/mod.rs (hunk 1 above)
Comment (brayniac) [reply src/rust-util/pelikan/src/core/mod.rs]: Yeah, this is a little strange. What would we be blocking by rewriting now? Is there anything that blocks doing this better in this PR?

File: src/rust-util/pelikan/src/core/mod.rs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    fn read(
+        rbuf: &mut OwnedBuf,
+        wbuf: &mut OwnedBuf,
+        state: &mut *mut (),
+    ) -> Result<(), DataProcessorError>;
+    fn write(
+        rbuf: &mut OwnedBuf,
+        wbuf: &mut OwnedBuf,
+        state: &mut *mut (),
+    ) -> Result<(), DataProcessorError>;
```
Comment (brayniac) [inline src/rust-util/pelikan/src/core/mod.rs]: newline

**Pull request merged**